### DISCUSSION TITLE
Help with middleware on parent and child pages

### DISCUSSION THREAD

**Discussion by tm0nt** [nuxt/nuxt#25725] [General]
I'm starting with nuxt3 and I have a question.

I need to block access to some pages.

Example:

I have two user types user with permission = 1 and permission = 0;

Each permission requires a different access page. To prevent users from viewing pages that do not belong to their specific ID.

I need to block them.

I already have the pages in

/profile/1/*
/profile/0/*

I have already separated the pages for each specific userId into folders.

PS: If there was a way to lock all folders/files/children of the main folder, it would help and save time

**Comment by AloisSeckar** [comment]
File `/middleware/lock.global.ts`:

```
export default defineNuxtRouteMiddleware((to, from) => {
  if (to.path.includes('/profile/1') {
    if ( /* check if permission !== 1 */ ) {
       return // either abortNavigation or navigateTo
    }
  }
  if (to.path.includes('/profile/0') {
    if ( /* check if permission !== 0 */ ) {
       return // either abortNavigation or navigateTo
    }
  }
})
```

This will run out of the box before EACH routing and take care of respective routes based on `to.path`, which contains the target url (you can also use `.fullPath` property for complete url including query params after hash)

For more fine-grained approach, you can have separate middleware files (without `.global` in their names) and inside your pages' setup, you can associate them as needed via:

```
definePageMeta({
  middleware: ["middleware-name"]
  // or middleware: 'middleware-name'
})
```
